client/assets/javascripts/modules_new/DetailsExpander.js

```diff
@@ -1,27 +1,32 @@
 const DetailsExpander = {
   init: function (nodes) {
-    // TODO refactor to event delegation
-    const container = nodes.querySelector('.js-details-expander')
+    const container = document.querySelectorAll('[class^="js-details-expander"],[class*=" js-details-expander"]')
 
-    const expandDetails = function (event) {
-      const textareaGroup = container.querySelector('.js-details-expandable')
+    container.forEach((userItem) => {
+      const inputBox = userItem.querySelector('input[type="text"]')
+
+      inputBox.addEventListener('input', this.expandDetails)
+      inputBox.addEventListener('paste', this.expandDetails)
+      inputBox.addEventListener('change', this.expandDetails)
+    })
+  },
+
+  expandDetails: function (event) {
+    const container = document.querySelectorAll('[class^="js-details-expander"],[class*=" js-details-expander"]')
```

Comment:
As above re `document.querySelectorAll('.js-details-expander')` will pick up any element with that class anywhere in its class attributes.